src/batch/src/executor/delete.rs

```diff
@@ -99,30 +103,54 @@ impl DeleteExecutor {
         let mut notifiers = Vec::new();
 
         // Transform the data chunk to a stream chunk, then write to the source.
-        let write_chunk = |chunk: DataChunk| async {
+        let write_txn_data = |chunk: DataChunk| async {
             let cap = chunk.capacity();
             let stream_chunk = StreamChunk::from_parts(vec![Op::Delete; cap], chunk);
 
             self.dml_manager
-                .write_chunk(self.table_id, self.table_version_id, stream_chunk)
+                .write_txn_msg(
+                    self.table_id,
+                    self.table_version_id,
+                    TxnMsg::Data(self.txn_id, stream_chunk),
+                )
                 .await
         };
 
+        notifiers.push(
+            self.dml_manager
+                .write_txn_msg(
+                    self.table_id,
+                    self.table_version_id,
+                    TxnMsg::Begin(self.txn_id),
+                )
+                .await?,
+        );
+
         #[for_await]
         for data_chunk in self.child.execute() {
             let data_chunk = data_chunk?;
             if self.returning {
                 yield data_chunk.clone();
             }
             for chunk in builder.append_chunk(data_chunk) {
-                notifiers.push(write_chunk(chunk).await?);
+                notifiers.push(write_txn_data(chunk).await?);
             }
         }
```

Comment:
What happens if the `child.execute()` goes wrong? I am afriad that the `TxnMsg::End` will never be emitted, which might result in a forever dangling transaction

Reply:
```
Barrier => 
      if !active_txn_map.is_empty() {
          for (txn_id, (_, mut vec)) in active_txn_map
              .drain_filter(|_, (epoch, _)| epoch.0 < barrier.epoch.prev)
          {
              tracing::warn!("txn_id={}. In case of late `TxnMsg::End`, drain those transaction chunks", txn_id);
              for chunk in vec.drain(..) {
                  yield Message::Chunk(chunk);
              }
          }
      }
```

We expected that `TxnMsg::End` would be received for each small transaction within a maximum of 2 epochs. If the unexpected case occurs, send corresponding chunks to the downstream.

Reply:
@BugenZhao I found this branch could happen frequently, so let me consider removing it.

Reply:
I have removed that part because we guarantee a transaction always writes its data to the same channel and it will send a `rollback` message to the `DmlExecutor` if there an error occurs, so there should be no dangling transaction by design.